tests/baselines/reference/objectLiteralGettersAndSetters.symbols

```diff
@@ -40,8 +40,8 @@ var sameName4a = { get ''() { return ''; }, set ""(n) { var p = n; var p: string
 
 var sameName5a = { get '\t'() { return ''; }, set '\t'(n) { var p = n; var p: string; } };
 >sameName5a : Symbol(sameName5a, Decl(objectLiteralGettersAndSetters.ts, 5, 3))
->'\t' : Symbol('\t', Decl(objectLiteralGettersAndSetters.ts, 5, 18), Decl(objectLiteralGettersAndSetters.ts, 5, 45))
->'\t' : Symbol('\t', Decl(objectLiteralGettersAndSetters.ts, 5, 18), Decl(objectLiteralGettersAndSetters.ts, 5, 45))
+>'\t' : Symbol('t', Decl(objectLiteralGettersAndSetters.ts, 5, 18), Decl(objectLiteralGettersAndSetters.ts, 5, 45))
+>'\t' : Symbol('t', Decl(objectLiteralGettersAndSetters.ts, 5, 18), Decl(objectLiteralGettersAndSetters.ts, 5, 45))
 >n : Symbol(n, Decl(objectLiteralGettersAndSetters.ts, 5, 55))

```

Comment:
The code to remove the `\` dates back a long long time.
```ts
symbolName.substring(1, symbolName.length - 1).replace(/\\./g, s => s.substring(1))
```
 I have traced it back to this commit https://github.com/microsoft/TypeScript/pull/18860/commits/8bb7230729f43f1c134f895a535e73c4cd58af3e 

Maybe @weswigham can shed more light on it, but this seems to be there to deal with escape sequences. Since when we build the string we need the escaped version, but what we have in `symbolName` is the source code version of the string. I will look for a better solution. 

The current version can cause issues in both quick info and declaration emit:

```ts
// index.ts
class T {
    static readonly "\t" = Symbol()
}

let x = { [T['\t']]: 1 }

//index.d.ts
declare class T {
    static readonly "\t": unique symbol;
}
declare let x: {
    [T["t"]]: number; // ❌ wrong symbol
};
```
[Playground Link](https://www.typescriptlang.org/play/?#code/MYGwhgzhAEAq0G8BQ1XQgFzBglsaATgKZgAmA9gHYgCe0ARADob3QC80AyjQLYBG5EAAoAlEgC+SJCCIZoAD3aJoAbVgqA5Mw0BdHQC5oARmjigA)